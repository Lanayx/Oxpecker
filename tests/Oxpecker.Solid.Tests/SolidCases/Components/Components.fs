module Oxpecker.Solid.Tests.Cases.Components

open Fable.Core.JS
open Oxpecker.Solid

[<SolidComponent>]
let Component (getText: Accessor<string>) =
    h1(onClick = fun _ -> console.log(getText())) {
        getText()
    }

[<SolidComponent>]
let Test () =
    let getText, _ = createSignal "Hello"

    div() {
        Component(getText)
    }
