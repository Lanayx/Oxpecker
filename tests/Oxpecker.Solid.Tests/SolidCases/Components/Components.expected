import { some } from "./fable_modules/fable-library-js.4.21.0/Option.js";
import { createSignal } from "solid-js";

export function Component(getText) {
    return <h1 onClick={(_arg) => {
            console.log(some(getText()));
        }}>
        {getText()}
    </h1>;
}

export function Test() {
    const patternInput = createSignal("Hello");
    return <div>
        {Component(patternInput[0])}
    </div>;
}

